### PR TITLE
8.x 2.x merge

### PR DESCRIPTION
**Pull request by damiankloip**
Testing my big merge into outdated `8.x2.x` branch

This does a few things:

- Updates old `8.x-2.x` branch with latest code from `8.x-1.x`
- Removes dependency on REST module, so configuration, plugins, response subscriber and friends
- Implements new `ApiResource` and `FormatNegotiator` plugins
- Refactors relevant code paths, like `ResourceController` to work with the new plugins instead
- It depends on replication `8.x-2.x` now, and the `replication.serializer` service.
- Uses core `/session/token` instead of the REST module equivalent. Deprecated in REST module anyway

This still needs some work, but probably better to iterate on what I have here from now on.

### CODE REVIEW

File: relaxed.module [hunk 1]
```diff
@@ -11,61 +11,16 @@ use Drupal\relaxed\Entity\Remote;
...
-  }
-}
-
-/**
  * Implements hook_entity_access().
  */
 function relaxed_entity_access(EntityInterface $entity, $operation, AccountInterface $account) {
-  /** @var \Drupal\workspace\WorkspaceManagerInterface $workspace_manager */
-  $workspace_manager = \Drupal::service('workspace.manager');
+  /** @var \Drupal\multiversion\MultiversionManagerInterface $manager */
```
Comment (jeqq) [inline relaxed.module]: Variable name is $multiversion_manager

File: relaxed.install [hunk 2]
```diff
@@ -19,8 +19,19 @@ function relaxed_install() {
...
+        'bypass node access',
       ],
     ])->save();
   }
 }
+
+/**
+ * Load new rest configuration for RESTful Web Services module.
+ */
+function relaxed_update_8301() {
```
Comment (damiankloip) [inline relaxed.install]: I think we should just remove this anyway? People upgrading would have already run this and REST config does not exist now with this work.

File: src/Controller/ResourceController.php [hunk 3]
```diff
@@ -168,106 +108,268 @@ public function errorResponse(\Exception $e) {
...
+   * This uses the content type format if an actual format is no specified.
+   *
+   * @return string
    */
-  public function csrfToken() {
-    return new Response(\Drupal::csrfToken()->get('rest'), 200, array('Content-Type' => 'text/plain'));
+  protected function getResponseFormat(Route $route, Request $request, ApiResourceInterface $api_resource) {
+    $api_resource_formats = $api_resource->getAllowedFormats();
+    $acceptable_response_formats = $route->hasRequirement('_format') ? explode('|', $route->getRequirement('_format')) : [];
+    //$acceptable_content_type_formats = $route->hasRequirement('_content_type_format') ? explode('|', $route->getRequirement('_content_type_format')) : [];
```
Comment (jeqq) [inline src/Controller/ResourceController.php]: Why do we have this commented code? Can it be removed?

**Review by jeqq — CHANGES_REQUESTED** [review]
I took a quick look, a lot of changes, I think it's in a good state as a start for 8.x-1.x. The only concern I have, is missing tests for the new implementations like negotiators, api classes, ...

Yes, and 2 comments for minor fixes.

**Comment by damiankloip** [comment]
@jeqq Yes, the new plugin manager logic needs some tests, the API resources do have coverage already with all the existing tests that have been converted though.

**Pull request merged**